include/oneapi/dpl/pstl/hetero/algorithm_ranges_impl_hetero.h

```diff
@@ -384,17 +384,17 @@ template <typename _BackendTag, typename _ExecutionPolicy, typename _Range1, typ
           typename _Assign = oneapi::dpl::__internal::__pstl_assign>
 oneapi::dpl::__internal::__difference_t<_Range2>
 __pattern_copy_if(__hetero_tag<_BackendTag> __tag, _ExecutionPolicy&& __exec, _Range1&& __rng1, _Range2&& __rng2,
-                  _Predicate __pred, _Assign)
+                  _Predicate __pred, _Assign&& __assign)
```

Comment:
In some places, `_Assign` is passed by copy and in others (such as here) it is passed by forwarding ref. Should we try to make these parameters consistent across the board?

Reply:
I think our convention is to just pass by copy elsewhere, I'll make this consistent for both assign and preds in this PR.

Reply:
done